driver/src/orderbook/streamed/state.rs

```diff
@@ -33,8 +45,31 @@ struct LastSolution {
 }
 
 impl State {
-    /// Can be used to create a `crate::models::AccountState`.
-    pub fn account_state(
+    /// Errors if State has only received a partial solution which would make the result
+    /// inconsistent.
+    /// Errors if the batch is `Future` but is not actually in the future.
+    /// Account balances that overflow a U256 are skipped.
+    pub fn orderbook(
+        &self,
+        batch: Batch,
+    ) -> Result<(
+        impl Iterator<Item = ((UserId, TokenId), U256)> + '_,
+        impl Iterator<Item = ModelOrder> + '_,
+    )> {
+        let batch_id = match batch {
+            Batch::Current => self.last_batch_id,
+            Batch::Future(batch_id) => {
+                ensure!(self.last_batch_id < batch_id, "batch is not in the future");
```

Comment:
with the strict inequality I'm a little worried about race conditions here, if we thought something was in a future, but happens to be `Current` now, I think that should be still fine.